glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -75,6 +107,102 @@ open class GleanInternalAPI internal constructor () {
         return initialized.toInt() != 0
     }
 
+    /**
+     * Register the pings generated from `pings.yaml` with Glean.
+     *
+     * @param pings The `Pings` object generated for your library or application
+     * by Glean.
+     */
+    fun registerPings(pings: Any) {
+        // Instantiating the Pings object to send this function is enough to
+        // call the constructor and have it registered in [PingType.pingRegistry].
+        Log.i(LOG_TAG, "Registering pings for ${pings.javaClass.canonicalName}")
+    }
+
+    /**
+     * Enable or disable Glean collection and upload.
+     *
+     * Metric collection is enabled by default.
+     *
+     * When uploading is disabled, metrics aren't recorded at all and no data
+     * is uploaded.
+     *
+     * When disabling, all pending metrics, events and queued pings are cleared.
+     *
+     * When enabling, the core Glean metrics are recreated.
+     *
+     * @param enabled When true, enable metric collection.
+     */
+    fun setUploadEnabled(enabled: Boolean) {
+        LibGleanFFI.INSTANCE.glean_set_upload_enabled(handle, enabled.toByte())
+    }
+
+    /**
+     * Get whether or not Glean is allowed to record and upload data.
+     */
+    fun getUploadEnabled(): Boolean {
+        return LibGleanFFI.INSTANCE.glean_is_upload_enabled(handle) != 0.toByte()
+    }
+
+    /**
+     * Indicate that an experiment is running. Glean will then add an
+     * experiment annotation to the environment which is sent with pings. This
+     * information is not persisted between runs.
+     *
+     * @param experimentId The id of the active experiment (maximum
+     *     30 bytes)
+     * @param branch The experiment branch (maximum 30 bytes)
+     * @param extra Optional metadata to output with the ping
+     */
+    fun setExperimentActive(
+        experimentId: String,
+        branch: String,
+        extra: Map<String, String>? = null
+    ) {
+        Log.e(LOG_TAG, "setExperimentActive is a stub")
+        // TODO: stub
+        // ExperimentsStorageEngine.setExperimentActive(experimentId, branch, extra)
+    }
+
+    /**
+     * Indicate that an experiment is no longer running.
+     *
+     * @param experimentId The id of the experiment to deactivate.
+     */
+    fun setExperimentInactive(experimentId: String) {
+        Log.e(LOG_TAG, "setExperimentInactive is a stub")
+        // TODO: stub
```

Comment:
Let's please file a bug about this and add the bug number here (I think this can be the same bug as above).